### ISSUE TITLE
Consistent naming of extensions .r or .R

### ISSUE THREAD

**Issue by rho62** [ocbe-uio/TruncExpFam#31]
As I understand it, some files will be merged at the end, so perhaps immaterial for these. But we could just stick to one or the other. Which do you prefer

**Comment by wleoncio** [comment]
I agree that the file extensions should be standardized. I used lowercase (.r) because it's my editor's default (I was feeling a bit lazy), but I actually prefer .R

**Comment by rho62** [comment]
Hi Waldir

A couple of questions

How do I add labels to issues (like «Bug», «Help wanted» etc)?
Now I have two branches running. What happens with them, once you’ve merged them with the «develop» line? Can I delete them or does that happen automatically

/René

3. feb. 2021 kl. 10:10 skrev Waldir Leoncio <notifications@github.com<mailto:notifications@github.com>>:



Closed #31<https://github.com/ocbe-uio/TruncExpFam/issues/31> via c19d965<https://github.com/ocbe-uio/TruncExpFam/commit/c19d965ec854453f7bcfe6de37bb425ca28aa6d8>.

**Comment by wleoncio** [comment]
Hi René,

> How do I add labels to issues (like «Bug», «Help wanted» etc)?

I'm afraid adding labels (and assignees, projects, milestones) is restricted to internal collaborators to the repository. I can add you to the list as soon as you want (though you should accept that invitation to the ocbe-uio page, first).

> Now I have two branches running. What happens with them, once you’ve merged them with the «develop» line? Can I delete them or does that happen automatically

Branch deletion doesn't happen automatically on Git, so as a general rule one must explicitly delete the branch once it's merged. Thankfully, GitHub does create a button to do so after a PR is merged.

As a side note, a branch is just a pointer to a certain commit in the repository, so deleting branches is normally a safe procedure, as it doesn't delete the commit history.

# Sample workflow

For the record, here's a suggestion of a workflow between your local terminal and your fork.

```bash
git fetch  # makes sure your repository sees the latest version of your fork
git pull # syncs the content of your repository with your fork
# make changes to the relevant files in the file editor of your preference
git add -A # stages all changed files to be committed
git commit -m "Fixes Issue #100" # commit changes with a message
git push # sync the changes back to your fork
```

Adding that I often use `git status` and `git log` in between those commands to make sure the files are really moving to the places I intend them to.

After your fork is updated, you can go to its homepage and there should be a button for issuing a pull request.

# Reference

The sample workflow above is based on [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/), a workflow model for Git which is well-worth a read!

**Comment by wleoncio** [comment]
I think I'm gonna add this sample workflow to the Wiki for visibility, might be useful for other potential collaborators and future projects 🤔 

Edit: It's up: https://github.com/ocbe-uio/TruncExpFam/wiki/Contributing-to-the-project

**Comment by rho62** [comment]
Hi Waldir

Hadn’t paid attention to the invitation at github. And now it seems to have expired. Can I ask you to re-issue it? Thx
/René



3. feb. 2021 kl. 11:20 skrev Waldir Leoncio <notifications@github.com<mailto:notifications@github.com>>:



Hi René,

How do I add labels to issues (like «Bug», «Help wanted» etc)?

I'm afraid adding labels (and assignees, projects, milestones) is restricted to internal collaborators to the repository. I can add you to the list as soon as you want (though you should accept that invitation to the ocbe-uio page, first).

Now I have two branches running. What happens with them, once you’ve merged them with the «develop» line? Can I delete them or does that happen automatically

Branch deletion doesn't happen automatically on Git, so as a general rule one must explicitly delete the branch once it's merged. Thankfully, GitHub does create a button to do so after a PR is merged.

As a side note, a branch is just a pointer to a certain commit in the repository, so deleting branches is normally a safe procedure, as it doesn't delete the commit history.

Sample workflow

For the record, here's a suggestion of a workflow between your local terminal and your fork.

git fetch  # makes sure your repository sees the latest version of your fork
git pull # syncs the content of your repository with your fork
# make changes to the relevant files in the file editor of your preference
git add -A # stages all changed files to be committed
git commit -m "Fixes Issue #100" # commit changes with a message
git push # sync the changes back to your fork

Adding that I often use git status and git log in between those commands to make sure the files are really moving to the places I intend them to.

After your fork is updated, you can go to its homepage and there should be a button for issuing a pull request.

Reference

The sample workflow above is based on GitFlow<https://nvie.com/posts/a-successful-git-branching-model/>, a workflow model for Git which is well-worth a read!

**Comment by rho62** [comment]
Hi Waldir

Two more issues/questions

1) I’ve made a pull, after which I expected the content on my local folder to be similar to that in the «develop» folder; e.g the .r and .R thing. But it’s not
2) I’m not sure where the dispatching takes place, i.e that rtrunc knows which distribution I aim to sample from. I’d like to update the rtrunc.R file, but want to be certain about the structure first.

Do you have for a zoom tomorrow morning, say 9am?. It shouldn’t take long

/René

3. feb. 2021 kl. 11:21 skrev Waldir Leoncio <notifications@github.com<mailto:notifications@github.com>>:



I think I'm gonna add this sample workflow to the Wiki for visibility, might be useful for other potential collaborators and future projects 🤔

**Comment by wleoncio** [comment]
Hi René,

> Hadn’t paid attention to the invitation at github. And now it seems to have expired. Can I ask you to re-issue it? Thx

Done! 👍🏽 

> 1) I’ve made a pull, after which I expected the content on my local folder to be similar to that in the «develop» folder; e.g the .r and .R thing. But it’s not

Did you remember to use `git fetch` before the pull? 

> 2) I’m not sure where the dispatching takes place, i.e that rtrunc knows which distribution I aim to sample from. I’d like to update the rtrunc.R file, but want to be certain about the structure first.

Dispatching is done based on the combination of arguments passed, which is registered in the `signature` argument of the generic and the method functions. Not sure this answers your questions, but we can go over it in our next meeting.

**Warning**: I wouldn't worry too much about the structure of rtrunc right now; it could still suffer major changes as I work on a fix for issue #18. I've posted a [question on StackOverflow](https://stackoverflow.com/q/66024978/1169233) yesterday regarding this and someone posted a solution that uses S3, so it's still possible we switch back to it.
